.github/workflows/tests.yaml

```diff
@@ -0,0 +1,236 @@
+name: NATS Server Tests
+
+on:
+  # push:
+  # pull_request:
+  schedule:
+    - cron: "0 * * * *" # Hourly
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ${{ vars.GHA_WORKER_SMALL }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: stable
```

Comment:
This means that when a new Go minor releases, based on past experience we risk having the lint tests failing until 
 1. main is updated with new test data for certs or whatever else they've changed in crypto/tls for the default assumption of the-world-is-http
 2. the lint tools are updated to handle new syntax elements

And similarly for where the matrix is `[ stable, oldstable ]`.

It's fine (for `main`, not release branches), as long as you're _expecting_ this to periodically break and don't want this to be a var?

Reply:
Good point, I think the best compromise here is to have two build steps: one that tests against latest (as our users will obviously do the same) and the other that tests against whatever our minimum supported version is set to in `go.mod`. That should mean that we can bump the minimum version without having to mess with the workflow.